### ISSUE TITLE
macOS theme(s)?

### ISSUE THREAD

**Issue by 0rAX0** [KeitIG/museeks#157]
Hey, nice player! I've come to know about it just minutes ago! :)

Would you be interested in having a better UI integration in macOS? I might be able to help design and theme your app in macOS but I need help.

**Comment by YurySolovyov** [comment]
See #153

**Comment by KeitIG** [comment]
@0rAX0 Thanks for your interest !

By "a better UI integration", do you mean "the main frame" or more generally all the UI components ?

**Comment by YurySolovyov** [comment]
@0rAX0 
> but I need help.

what kind of help? If you can do mock-ups that's great too!

**Comment by 0rAX0** [comment]
@KeitIG I meant a better UI for macOS, one that looks better next to other apps. Currently, it looks like a Like (GTK) app. That could lead to enhancing all the UI components.

@YurySolovyov For starters, I need to know if there is a way to add a new stylesheet to experiment on, and if there is a way to see the changes as I edit them. I've done theming before (GNOME Shell and Firefox) but this is new for me. Unless you want to handle the coding.

And yes, before starting I have to have a mockup.

**Comment by YurySolovyov** [comment]
@0rAX0 take a look at readme, you need to start dev server that would watch for style and component changes.

Also, for relatively quick feedback loop you may consider using prebuild `electron` package like so:
```sh
# in cloned repo
npm i -g electron
npm i
electron .
```
Also, you can always find us on gitter for quicker interaction.

**Comment by 0rAX0** [comment]
OK, thanks! I'll take a look and get back to you.

**Comment by YurySolovyov** [comment]
@0rAX0 there is also DevMode checkbox in Advanced settings panel that opens up Chrome DevTools

**Comment by 0rAX0** [comment]
@YurySolovyov Yeah, I've seen that. 👍

**Comment by KeitIG** [comment]
Just to clarify some things: stylize components to make them look a bit more "mac-ish" is not something I'd like now. Atom does not do it and it's perfectly fine. It would be too hard to maintain some OS specific stylesheets. 

But the native frame inclusion in discussion there: #153

**Comment by 0rAX0** [comment]
@KeitIG Fair enough. I wouldn't have reported the issue if Museeks had its own theme like Atom has, but it did look a bit too Linux-y. :)

**Comment by YurySolovyov** [comment]
We might choose to use material icons, so I guess Material Design in general is a good inspiration source, but we might want to come up with museeks own style that does not directly follow Material

**Comment by KeitIG** [comment]
That's a nice observation, I'm not really objective about it. If you feel the envy to provide a more agnostic screenshot, please do so !

I'll close this for now. But the discussion is still open.

**Comment by 0rAX0** [comment]
## First try: 

I thought since you didn't want platform-specific sheets and wanted something unique to Museeks that I'd do something like this:

- It uses Material icons and Roboto (personal preference, Open Sans is fine) – I would have used Apple's Sans Fransisco to emphasize loading default system fonts if had access to it.
- I think it's unique enough for it not to be associated with any particular platform.
- It also has some fixes that I think the current layout needs, like in the _Preferences_ tab.
- Only the half-dark theme is used here, I can propose different colors if you like.

Feel free to rip it apart, it's just something to get the discussion going about the style, so what do you think?

![main halfdark](https://cloud.githubusercontent.com/assets/1044370/17812893/f98d53fc-6620-11e6-9742-c4267c990e86.png)
![preferences halfdark](https://cloud.githubusercontent.com/assets/1044370/17812894/f9931094-6620-11e6-9c2b-ec50b270f666.png)

**Comment by YurySolovyov** [comment]
Wow, that looks interesting.
Some notes:
1. We are planning to add minimize and maximize buttons, so that should be present
2. Need both full-dark and full-light themes
3. Queue and volume buttons look a bit too heavy, maybe you can drop outlines or something. Same for bottom buttons

**Comment by YurySolovyov** [comment]
Also, I feel like rounded borders bring weight too, and I'd go for simpler forms

**Comment by YurySolovyov** [comment]
One more thing: headers UI should look close to native counterparts at least on macOS and Win in terms of buttons placement, on linux, there is more freedom, but I'd say it is better to be closer to current placement

**Comment by 0rAX0** [comment]
@YurySolovyov 
1. Sure
2. I wanted to show you what I have so far, adding those is easy.
3. What do you mean by heavy and "bring weight"?

**Comment by YurySolovyov** [comment]
@0rAX0 I mean it brings visual noise

**Comment by 0rAX0** [comment]
OK, here:

I don't like those icons hovering there without border, though.

![main halfdark](https://cloud.githubusercontent.com/assets/1044370/17831459/1469799c-66e1-11e6-9b92-a706e19ae062.png)

**Comment by YurySolovyov** [comment]
IMO this a progress.

Maybe it is a matter of taste, but I like this better.

One thing to experiment is to try making dark color less saturated and lighter to make it closer to plain gray instead of blue, so that it looks a bit more [`flat`](http://flatuicolors.com/).

**Comment by KeitIG** [comment]
Hey,

I'll be honest, but don't take it wrong ;) I'll be more interested by UX improvements than UI improvements atm. There are good things in these mockups, but I won't change drastically the UI now. One last things: the light theme is light, and the dark one is dark, I'm not a fan of mixed things.

note to myself:
- `container` in the admin instead of `container-fluid` ?
- `SettingsLibrary` view needs a review #135 
- more aerated `TracksList` ?

**Comment by YurySolovyov** [comment]
@0rAX0 
While I agree with @KeitIG, I think that design is a continuous process and it will last as long as we have museeks going. I really appreciate your interest and your efforts.
Hope you can leave some feedback in #135 and #153.
Also, feel free to open new issues if you feel that design and UX can be improved in some areas. (kinda obvious advice, but still :) )

**Comment by 0rAX0** [comment]
@KeitIG This thread is about theming the app. I originally proposed that I help make the app look like a native one under macOS, you said that's not a goal of yours (very understandable) @YurySolovyov said that you're looking for a unique look so I thought why not propose something while I'm here to help you a bit?

I know that I've changed and moved things around but the focus should be on the theme i.e. colors, borders, and more colors. You seem to have been distracted by the UI changes. :)
Once everything is set, I'll split the design into a dark and a light theme; I can't do it now so I don't have multiple screens to maintain.

**TL;DR** While I have included lots of small changes here and there, my goal was to produce a good looking theme, nothing more and nothing less.

**Comment by KeitIG** [comment]
Poke #87

I'll create an issue for custom thèmes too.

**Comment by KeitIG** [comment]
See #162
